ansible_collections/arista/avd/tests/unit/modules/test_inventory_to_container.py

```diff
@@ -131,80 +134,80 @@ def test_is_leaf_none_leaf(self) -> None:
         assert output
         assert isinstance(output, dict)
 
-    def test_get_device_option_value_invalid(self, inventory) -> None:
+    def test_get_device_option_value_invalid(self, inventory: dict) -> None:
         data = inventory["all"]["children"]["CVP"]["hosts"]
         output = get_device_option_value(device_data_dict=data, option_name="is_deployed")
         assert output is None
 
-    def test_get_device_option_value_none(self, inventory) -> None:
+    def test_get_device_option_value_none(self, inventory: dict) -> None:
         data = inventory["all"]["children"]["CVP"]["hosts"]
-        output = get_device_option_value(device_data_dict=data, option_name=None)
+        output = get_device_option_value(device_data_dict=data, option_name=None)  # pyright: ignore[reportArgumentType]
         assert output is None
 
-    def test_get_device_option_value_empty_data(self, inventory) -> None:
-        output = get_device_option_value(device_data_dict=None, option_name="cv_server")
+    def test_get_device_option_value_empty_data(self) -> None:
+        output = get_device_option_value(device_data_dict=None, option_name="cv_server")  # pyright: ignore[reportArgumentType]
         assert output is None
 
     def test_get_devices_empty_inventory(self) -> None:
         output = get_devices(None)
         assert output is None
 
-    def test_get_devices_default_search_container(self, inventory) -> None:
+    def test_get_devices_default_search_container(self, inventory: dict) -> None:
         output = get_devices(inventory)
         assert output is None
 
-    def test_get_devices_non_default_search_container(self, inventory) -> None:
+    def test_get_devices_non_default_search_container(self, inventory: dict) -> None:
         output = get_devices(inventory, search_container=SEARCH_CONTAINER, devices=[])
         assert output == GET_DEVICES
 
-    def test_get_devices_preexisting_devices(self, inventory) -> None:
+    def test_get_devices_preexisting_devices(self, inventory: dict) -> None:
         devices = ["TEST_DEVICE"]
         output = get_devices(inventory, search_container=SEARCH_CONTAINER, devices=devices)
         assert output == ["TEST_DEVICE", *GET_DEVICES]
 
-    def test_get_devices_preexisting_devices_with_device_filter(self, inventory) -> None:
+    def test_get_devices_preexisting_devices_with_device_filter(self, inventory: dict) -> None:
         output = get_devices(inventory, search_container=SEARCH_CONTAINER, devices=[], device_filter=[GET_DEVICE_FILTER])
         assert [GET_DEVICE_FILTER in item for item in output]
 
-    @pytest.mark.parametrize("DATA", [None])
-    def test_serialize_empty_inventory(self, DATA) -> None:
-        output = serialize(DATA)
+    @pytest.mark.parametrize("data", [None])
+    def test_serialize_empty_inventory(self, data: dict | None) -> None:
+        output = serialize(data)  # pyright: ignore[reportArgumentType]
```

Comment:
ugh - I get this but are we mistyping the serialize input as it seems to accept None and you are getting a pyright issue?
Also I would vote for removing the parametrize givent the test

Reply:
probably same wrong type hints for other places where you have added the pyright ignore?

Reply:
actually the type hint in `serialize` does _not_ match the input as it states it must be a `dict` but here the parametrized input is `dict|None`, and this test is only testing `None`

we are testing that if we give `None`, then the serialize function gives back `None`.

The return type hint of `serialize` is currently `Tree` and not `Tree | None` so a lot of things, from a type hint point of view are not really great here. The best thing would actually be to raise when we don't receive an iterable.